board/app/controllers/comment_controller.php

```diff
@@ -0,0 +1,79 @@
+<?php
+class CommentController extends AppController {
+
+const MAX_COMMENT_PER_PAGE = 5;
+   //Write a new comment
+    public function write() {
+        $thread_id = Param::get('thread_id');
+
+        $thread = Thread::get($thread_id);
+        $comment = new Comment();
+        $current_page = Param::get('page_next', 'write');
+
+        switch($current_page) { 
+            case 'write':
+            break;
+
+        case 'write_end':                
+            $comment->body = Param::get('body');
+            $comment->username = Param::get('username');
+            
+            try 
+            {            
+               $comment->write($comment, $thread_id);
+            } catch (ValidationException $e) {                    
+                $current_page = 'write';
+            }    
+
+        break;
+        default:
+            throw new NotFoundException("{$current_page} is not found");
+            break;
+        }
+    $this->set(get_defined_vars());
+    $this->render($current_page);   
+    }
+
+    
+    public function delete() { 
+        $comment = Comment::get(Param::get('comment_id'));
+        $this->set(get_defined_vars());
+        $comment->deleteComment();
+        $this->render('comment/delete');
```

Comment:
- the order of `$this->set()` and `$comment->deleteComment()` is incorrect po
- method declarations must have it's own line, thus `{` should be in the next line po
- `deleteComment()` seems redundant since you are already calling the method from `$comment`, just `delete()` is fine po